### DISCUSSION TITLE
Sorting with tie-breakers

### DISCUSSION THREAD

**Discussion by irisjae** [directus/directus#17438] [Archive]
### Describe the Bug

Hi, I'm using Directus to administer a database. Certain rows in my database are being created in the same transaction with the same timestamp. However, this leads to very confusing behaviour in Directus, because it doesn't seem to provide tie-breakers when I'm sorting by timestamp.

As multiple rows have the same sorting value, and thus the order that rows appear is random. I see the same row across multiple pages, wheras other rows don't show up in any page.

### To Reproduce

Insert multiple rows in the same table with the same sorting value.

e.g.

```
create table example (tie_breaker integer primary key, timestamp timestamptz);
insert into example (tie_breaker, timestamp) values
 (1, now ())
, (2, now ())
, (3, now ())
, (4, now ())
, (5, now ())
, (6, now ())
, (7, now ())
, (8, now ())
, (9, now ())
, (10, now ())
, (11, now ())
, (12, now ())
, (13, now ())
, (14, now ())
, (15, now ())
, (16, now ())
, (17, now ())
, (18, now ())
, (19, now ())
, (20, now ())
, (21, now ())
, (22, now ())
, (23, now ())
, (24, now ())
, (25, now ())
, (26, now ())
, (27, now ())
, (28, now ())
, (29, now ())
, (30, now ())
, (31, now ())
, (32, now ())
, (33, now ())
, (34, now ())
, (35, now ())
, (36, now ())
, (37, now ())
, (38, now ())
, (39, now ())
, (40, now ())
, (41, now ())
, (42, now ())
, (43, now ())
, (44, now ())
, (45, now ())
, (46, now ())
, (47, now ())
, (48, now ())
, (49, now ())
, (50, now ())
, (51, now ())
, (52, now ())
, (53, now ())
, (54, now ())
, (55, now ())
, (56, now ())
, (57, now ())
, (58, now ())
, (59, now ())
, (60, now ())
, (61, now ())
, (62, now ())
, (63, now ())
, (64, now ())
, (65, now ())
, (66, now ())
, (67, now ())
, (68, now ())
, (69, now ())
, (70, now ())
, (71, now ())
, (72, now ())
, (73, now ())
, (74, now ())
, (75, now ())
, (76, now ())
, (77, now ())
, (78, now ())
, (79, now ())
, (80, now ())
, (81, now ())
, (82, now ())
, (83, now ())
, (84, now ())
, (85, now ())
, (86, now ())
, (87, now ())
, (88, now ())
, (89, now ())
, (90, now ())
, (91, now ())
, (92, now ())
, (93, now ())
, (94, now ())
, (95, now ())
, (96, now ())
, (97, now ())
, (98, now ())
, (99, now ())
, (100, now ());
```

### What version of Directus are you using?

9.14.1

### What version of Node.js are you using?

16.17.0

### What database are you using?

PostgreSQL 12

### What browser are you using?

Firefox

### How are you deploying Directus?

Running locally

**Comment by paescuj** [comment]
How do you even add this table to Directus as it doesn't contain a primary key?
Have you already looked at the built-in sort support via the [sort field](https://docs.directus.io/configuration/data-model/collections.html#sort-field)?

**Reply by irisjae** [reply]
My original table did contain a primary key, and I omitted most irrelevant details for illustration. I've added back the primary key as suggested, for completeness sake.

I've looked at the 'Manual sort' option, and I don't think it solves this use case. Our use case does not require any manually crafted sort; sometimes it may be useful to sort on some other field instead as well. Our only issue is that we'd like to let our staff be able to view all records when navigating through the pages.

I understand that without tie-breakers, SQL makes no guarantee for orderings on tie. If Directus would let us to for example, specify to sort on multiple columns (order by column_1, column_2) to break ties, that would solve our issue.
